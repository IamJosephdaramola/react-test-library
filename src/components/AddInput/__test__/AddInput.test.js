import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import AddInput from "../AddInput";

const mockedSetTodos = jest.fn();

describe("AddInput", () => {
  afterEach(cleanup);

  test("should take a snapshot of the input", () => {
    const { asFragment } = render(<AddInput />);
    expect(asFragment(<AddInput />)).toMatchSnapshot();
  });

  test("should render AddInput element", () => {
    render(<AddInput todos={[]} setTodos={mockedSetTodos} />);

    const inputElement = screen.getByPlaceholderText(/Add a new task here.../i);

    expect(inputElement).toBeInTheDocument();
  });

  test("should render be able to type into input", () => {
    render(<AddInput todos={[]} setTodos={mockedSetTodos} />);

    const inputElement = screen.getByPlaceholderText(/Add a new task here.../i);

    fireEvent.change(inputElement, { target: { value: "Go home by 6pm" } });

    expect(inputElement.value).toBe("Go home by 6pm");
  });

  test("should have empty input when add button is clicked", () => {
    render(<AddInput todos={[]} setTodos={mockedSetTodos} />);

    const inputElement = screen.getByPlaceholderText(/Add a new task here.../i);
    const buttonElement = screen.getByRole("button");

    fireEvent.change(inputElement, { target: { value: "Go home by 6pm" } });
    fireEvent.click(buttonElement);

    expect(inputElement.value).toBe("");
  });
});
