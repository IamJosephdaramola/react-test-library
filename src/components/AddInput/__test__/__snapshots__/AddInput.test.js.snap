// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddInput should take a snapshot of the input 1`] = `
<DocumentFragment>
  <div
    class="input-container"
  >
    <input
      class="input"
      placeholder="Add a new task here..."
      value=""
    />
    <button
      class="add-btn"
    >
      Add
    </button>
  </div>
</DocumentFragment>
`;
